services/user/index.js

```diff
@@ -1,9 +1,32 @@
-export const findUserByEmail = async (email) => {}
+import { userModel } from "../../db/models/user.js";
+import { ApiError } from "../../utils/ApiError.js";
 
-export const createUser = async (user) => {}
+/**
+ *
+ * @param {string} email - to unique identifier of the user
+ * @returns {Promise<userModel>} - to resolve to the find user
+ * @throws {Error} - if error occured in finding user.
+ */
 
-export const findUserById = async (id) => {}
+export const findUserByEmail = async (email) => {
```

Comment:
this could be used as a mongoose methods and static. you can have a method to fetch users by email, but do not write the condition to check if it exists or not in that method. Do it wherever you're calling it

Reply:
Seems like this has been misunderstood. 
I meant to have a static/method in the user model called `findByEmail`, which doesn't throw any error just returns User or null. 

You can then check for null status and throw an error.